target/linux/ramips/dts/HC5962.dts

```diff
@@ -0,0 +1,154 @@
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "HiWiFi HC5962";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x10000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		status {
+			label = "hc5962:white:status";
+			gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
+		};	
+		
+		system {
+			label = "hc5962:red:system";
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 18 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partition@0 {
+		label = "u-boot";
+		reg = <0x0 0x80000>;
+		read-only;
+	};
+
+	partition@80000 {
+		label = "debug";
+		reg = <0x80000 0x80000>;
+		read-only;
+	};
+
+	factory: partition@100000 {
+		label = "factory";
+		reg = <0x100000 0x40000>;
+		read-only;
+	};
+
+	partition@140000 {
+		label = "kernel";
+		reg = <0x140000 0x200000>;
+	};
+
+	partition@340000 {
+		label = "ubi";
+		reg = <0x340000 0x1E00000>;
+	};
+
+	partition@2140000 {
```

Comment:
If this partition is required to revert to the OEM firmware, mark the partition as read-only. If this partition is not required, drop the node and use the extra space for the ubi partition.